plugins/permission-node/src/ServerPermissionClient.ts

```diff
@@ -74,23 +78,34 @@ export class ServerPermissionClient implements PermissionsService {
       auth,
       permissionClient,
       permissionEnabled,
+      pluginId: options.pluginId,
     });
   }
 
   private constructor(options: {
     auth: AuthService;
     permissionClient: PermissionClient;
     permissionEnabled: boolean;
+    pluginId?: string;
   }) {
     this.#auth = options.auth;
     this.#permissionClient = options.permissionClient;
     this.#permissionEnabled = options.permissionEnabled;
+    this.#pluginId = options.pluginId;
   }
 
   async authorizeConditional(
     queries: QueryPermissionRequest[],
     options?: PermissionsServiceRequestOptions,
   ): Promise<PolicyDecision[]> {
+    const maybeResponse = this.#decideBasedOnPrincipalAccessRestrictions(
```

Comment:
Feeling it might be a bit cleaner to move this down to replace the `queries.map(_ => ({ result: AuthorizeResult.ALLOW }))` logic? Having that be the service principal path where we decide based on the access restrictions. Feels a bit strange that we're checking for the service principal twice right now.

Reply:
The reason I made it a standalone method that's done at the top, is that I wanted to retain the old fallback functionality perfectly (which ALSO in a way has logic for this token type). Thus, when no restrictions exist, it'll just keep running the old code path. Would you prefer I update that?

Reply:
ok @Rugvip tried for a bit of a refactor - but the way that things are intertwined, am not sure it got all that much better 🤔